### PR TITLE
add load tester suite

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/tester/suite.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func NewUnitSuite(t *testing.T) *unitSuite {
 }
 
 type unitSuite struct {
+	//nolint:forbidigo
```
Comment (ashmrtn) [inline src/internal/tester/suite.go:56]: these shouldn't be required as there's an exclude rule in .golangci.yaml file for it

File: src/pkg/repository/loadtest/repository_load_test.go [hunk 2]
```diff
@@ -384,38 +384,52 @@ func normalizeCategorySet(t *testing.T, cats map[string]struct{}) []string {
 	return sl
 }
 
+/* ================================================
+*   A note on load test setup:
+*   Even though most of the code here is boiler-
+*   plate and could be easily compressed into a
+*   test matrix, we want to keep the suites separte
```
Comment (ashmrtn) [inline src/pkg/repository/loadtest/repository_load_test.go:391]:
```suggestion
*   test matrix, we want to keep the suites separate
```

File: src/pkg/repository/loadtest/repository_load_test.go [hunk 3]
```diff
@@ -384,38 +384,52 @@ func normalizeCategorySet(t *testing.T, cats map[string]struct{}) []string {
...
 }
 
-func TestRepositoryLoadTestExchangeSuite(t *testing.T) {
-	tester.RunOnAny(t, tester.CorsoLoadTests)
-	suite.Run(t, new(RepositoryLoadTestExchangeSuite))
+func TestLoadExchangeSuite(t *testing.T) {
+	suite.Run(t, &LoadExchangeSuite{
+		Suite: tester.NewLoadSuite(
+			t,
+			[][]string{tester.AWSStorageCredEnvs, tester.M365AcctCredEnvs},
```
Comment (ashmrtn) [inline src/pkg/repository/loadtest/repository_load_test.go:416]: should we just make this part of the default check for the load suite in tester package?

**Review by ashmrtn — APPROVED** [review]

File: src/internal/tester/suite.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/tester/suite.go:34]: s'what I get for going by the ide and not the lint command.

File: src/pkg/repository/loadtest/repository_load_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/pkg/repository/loadtest/repository_load_test.go:417]: not necessarily.  It's a nominal addition, and we may want this granular control for local testing or if/when we introduce non-microsoft platforms.

**Pull request merged**